jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java

```diff
@@ -104,6 +104,18 @@ static Builder withFileName(String partAndFileName) {
      */
     Optional<String> getFileName();
 
+    /**
+     * Get the content object from this part.
+     * <p>
+     * If the content is represented by an un-consumed {@link InputStream InputStream} the method will return the input
+     * stream
+     * </p>
```

Comment:
The text here seems too vague to me, and I worry that it might confuse users.  IIUC, the intent of the `getContent` method is:
(1) on the _sending_ side, this method returns the actual object (`String`, `File`, `Widget`, `byte[]`, etc.) that was passed in to the EntityPart's builder.
(2) on the _receiving_ side, this method is basically the same as `getContentStream`, but requires the user to cast it.

I think this method really only has any value on the sending side.  Maybe we could change this to just throw an `IllegalStateException` on the receiving side?

Alternatively, we could view the EntityPart as a "holder" for the entity object on both the sending and receiving sides - then it should always return the entity object (`String`, `Widget`, etc.) but only after a (not yet written) `bufferContent()` is invoked followed by `getContent(Class)` - similar to `Response#bufferEntity()` and `Response#readEntity(...)`.

Reply:
Thanks for review @andymc12 
> I think this method really only has any value on the sending side. Maybe we could change this to just throw an IllegalStateException on the receiving side?

Do you mean this method is only for sending side and calling this method throws IllegalStateException on the receiving side ? 

> Alternatively, we could view the EntityPart as a "holder" for the entity object on both the sending and receiving sides - then it should always return the entity object (String, Widget, etc.) but only after a (not yet written) bufferContent() is invoked followed by getContent(Class) - similar to Response#bufferEntity() and Response#readEntity(...).

+1 . I think this is a good idea.  Should we still keep 'getContentStream()' ?